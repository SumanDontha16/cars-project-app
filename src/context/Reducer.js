/* eslint import/no-anonymous-default-export: [2, {"allowArrowFunction": true}] */
export default (state, action) => {
	switch (action.type) {
		case "CREATE_NEW_CAR":
			return {
				...state,
				car: [...state.car, action.payload],
			};

		case "UPDATE_CAR":
			const editingCar = action.payload;
			const updatedCar = state.car.map((car) => {
				if (car.id === editingCar.id) {
					return editingCar;
				}
				return car;
			});
			return {
				...state,
				car: updatedCar,
			};

		case "DELETE_CAR":
			return {
				...state,
				car: state.car.filter((car) => car.id !== action.payload),
			};

		default:
			return state;
	}
};
