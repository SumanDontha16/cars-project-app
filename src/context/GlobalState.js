import React, { createContext, useReducer } from "react";
import Reducer from "./Reducer";

const initialState = {
	car: [
		{
			id: 1,
			make: "Ford",
			model: "Puma",
			colour: "Black",
			year: "2009",
		},
	],
};

export const GlobalContext = createContext(initialState);
export const GlobalProvider = ({ children }) => {
	const [state, dispatch] = useReducer(Reducer, initialState);

	const createNewCar = (car) => {
		dispatch({
			type: "CREATE_NEW_CAR",
			payload: car,
		});
	};

	const deleteCar = (id) => {
		dispatch({
			type: "DELETE_CAR",
			payload: id,
		});
	};

	const updateCar = (car) => {
		dispatch({
			type: "UPDATE_CAR",
			payload: car,
		});
	};

	return (
		<GlobalContext.Provider
			value={{
				car: state.car,
				createNewCar,
				deleteCar,
				updateCar,
			}}
		>
			{children}
		</GlobalContext.Provider>
	);
};
