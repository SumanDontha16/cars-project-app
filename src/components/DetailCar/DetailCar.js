import React, { useState, useEffect, useContext } from "react";
import { Link, useParams } from "react-router-dom";
import { GlobalContext } from "../../context/GlobalState";

const DetailCar = () => {
	const [data, setData] = useState("");

	const { car } = useContext(GlobalContext);
	const { id } = useParams();

	useEffect(() => {
		const carId = id;
		const selectedCar = car.find((c) => c.id === parseInt(carId));
		setData(selectedCar);
		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, []);

	return (
		<>
			<div className="container">
				<div>
					<div className="row">
						<div className="col-md-12">
							<Link to="/">
								<button type="button" className="btn btn-dark">
									Go Back
								</button>
							</Link>
						</div>
					</div>
					<div className="listing">
						<div className="row">
							<div className="col-md-12">
								<span>{data && data.make}</span> (
								<span>{data && data.year}</span>)<h4>{data && data.model}</h4>
								<span>{data && data.colour}</span>
								<br />
								<br />
								<p className="lead">
									<small>
										Lorem Ipsum is simply dummy text of the printing and
										typesetting industry. Lorem Ipsum has been the industry's
										standard dummy text ever since the 1500s, when an unknown
										printer took a galley of type and scrambled it to make a
										type specimen book.
									</small>
								</p>
							</div>
						</div>
					</div>
				</div>
			</div>
		</>
	);
};

export default DetailCar;
