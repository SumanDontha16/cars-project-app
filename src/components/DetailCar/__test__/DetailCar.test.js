import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import DetailCar from "../DetailCar";
import { MemoryRouter } from "react-router-dom";
import { GlobalProvider } from "../../../context/GlobalState";

describe("Detail Car", () => {
	test("Component rendered successfully!", () => {
		const { container } = render(
			<GlobalProvider>
				<MemoryRouter>
					<DetailCar />
				</MemoryRouter>
			</GlobalProvider>
		);
		expect(container).toMatchSnapshot();
	});

	test("Go Back working successfully!", () => {
		render(
			<GlobalProvider>
				<MemoryRouter>
					<DetailCar />
				</MemoryRouter>
			</GlobalProvider>
		);

		const button = screen.getByRole("button", { name: "Go Back" });
		userEvent.click(button);
	});
});
