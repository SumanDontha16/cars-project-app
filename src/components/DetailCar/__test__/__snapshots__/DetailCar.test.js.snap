// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Car Component rendered successfully! 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <a
            href="/"
          >
            <button
              class="btn btn-dark"
              type="button"
            >
              Go Back
            </button>
          </a>
        </div>
      </div>
      <div
        class="listing"
      >
        <div
          class="row"
        >
          <div
            class="col-md-12"
          >
            <span />
             (
            <span />
            )
            <h4 />
            <span />
            <br />
            <br />
            <p
              class="lead"
            >
              <small>
                Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
              </small>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
